protocol/buff/reader.go

```diff
@@ -50,32 +50,30 @@ func (b *Message) Len() int {
 	return len(b.Bts)
 }
 
-// Discard skips the next n bytes.
-func (b *Message) Discard(n int) {
+// AssertAllocated panics if there aren't n bytes in the buffer.
+func (b *Message) AssertAllocated(n int) {
 	if len(b.Bts) < n {
 		panic("buffer overread")
```

Comment:
Would be great if this function is inlined. But AFAIK Go doesn't inline functions that can panic. I wonder if it does, can you check that?

And if it doesn't, maybe we can trick the compiler to do it anyways if we wrap `panic()` into our own helper function.

Reply:
It does inline this function.

```bash
$ go build -gcflags=-m github.com/edgedb/edgedb-go/protocol/codecs 2>&1 | grep AssertAllocated
protocol/codecs/array.go:73:13: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/array.go:77:14: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/array.go:81:13: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/base_scalar.go:142:21: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/base_scalar.go:215:21: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/base_scalar.go:259:13: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/object.go:109:13: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/object.go:112:14: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/object.go:118:15: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/object.go:121:16: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/set.go:68:13: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/set.go:72:14: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/set.go:76:13: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/tuple.go:88:13: inlining call to buff.(*Message).AssertAllocated
protocol/codecs/tuple.go:94:14: inlining call to buff.(*Message).AssertAllocated
```

Reply:
Cool, thanks for confirming